src/nbl/asset/utils/CIESProfile.cpp

```diff
@@ -0,0 +1,145 @@
+#include "CIESProfile.h"
+
+using namespace nbl;
+using namespace asset;
+
+const CIESProfile::IES_STORAGE_FORMAT& CIESProfile::getIntegral() const
+{
+    size_t numHSamples = 2 * hAngles.size();
+    size_t numVSamples = 2 * vAngles.size();
+    double dTheta = core::PI<double>() / numVSamples;
+    double dPhi = 2 * core::PI<double>() / numHSamples;
+    double dThetaInAngle = MAX_VANGLE / numVSamples;
+    double dPhiInAngle = MAX_HANGLE / numHSamples;
+    double res = 0;
+    for (size_t i = 0; i < numVSamples; i++) {
+        for (size_t j = 0; j < numHSamples; j++) {
+            auto val = dPhi * dTheta * std::sin(dTheta * i) * sample(i * dThetaInAngle, j * dPhiInAngle);
+            res += val;
+        }
+    }
+    return res;
+}
+
+const CIESProfile::IES_STORAGE_FORMAT& CIESProfile::sample(double vAngle, double hAngle) const {
```

Comment:
why are you returning a reference to a temporary!?

Reply:
resolved in following commit